addon/adapters/adapter.js

```diff
@@ -156,6 +195,82 @@ export default Adapter.extend(ImportExportMixin, {
     return null;
   },
 
+  _queryFilter(data, query = {}) {
+    const queryType = typeOf(query),
+      dataType = typeOf(data);
+
+    if (queryType === 'object' && dataType === 'object') {
+      return keys(query).every((key) => {
+        let queryValue = query[key],
+          recordValue;
+
+        // normalize type
+        if (key === 'type' && typeOf(queryValue) === 'string') {
+          queryValue = inflector.pluralize(queryValue);
+        }
+
+        // Attributes
+        if (key === 'id' || key === 'type') {
+          recordValue = data[key];
+        } else {
+          recordValue = data.attributes[key];
```

Comment:
Change to:
```
recordValue = data.hasOwnProperty('attributes') ?
           data.attributes[key] : null;
```
Otherwise it errors out if the record has no attributes.